### ISSUE TITLE
New version of Bote (2015_08_21)  throws an error (Java 1.6) (Trac #1644)

### ISSUE THREAD

**Issue by str4d** [i2p/i2p.i2p-bote#60]
I2P version: 0.9.21-0
Java version: Sun Microsystems Inc. 1.6.0_36 (OpenJDK Runtime Environment 1.6.0_36-b36)
Wrapper version: 3.5.15
Server version: 8.1.17.v20150415
Servlet version: Jasper JSP 2.1 Engine
Platform: Linux i386 3.2.0-23-generic
Processor: Athlon 5350 APU (athlon64)
Jbigi: Locally optimized native BigInteger library loaded from file
Encoding: UTF-8
Charset: UTF-8

Error is
HTTP ERROR 500

Problem accessing /i2pbote/index.jsp. Reason:

    Server Error

Caused by:

java.lang.IllegalArgumentException: Illegal pattern character 'X'
	at java.text.SimpleDateFormat.compile(SimpleDateFormat.java:769)
	at java.text.SimpleDateFormat.applyPattern(SimpleDateFormat.java:2023)
	at org.apache.taglibs.standard.tag.common.fmt.FormatDateSupport.doEndTag(FormatDateSupport.java:164)
	at org.apache.jsp.tag.web.printDate_tag._jspx_meth_fmt_005fformatDate_005f0(printDate_tag.java:254)
	at org.apache.jsp.tag.web.printDate_tag._jspx_meth_c_005fif_005f0(printDate_tag.java:170)
	at org.apache.jsp.tag.web.printDate_tag.doTag(printDate_tag.java:130)
	at i2p.bote.jsp.folder_jsp._jspx_meth_ib_005fprintDate_005f0(folder_jsp.java:2826)
	at i2p.bote.jsp.folder_jsp._jspx_meth_c_005fforEach_005f0(folder_jsp.java:1941)
	at i2p.bote.jsp.folder_jsp._jspx_meth_ib_005frequirePassword_005f0(folder_jsp.java:346)
	at i2p.bote.jsp.folder_jsp._jspService(folder_jsp.java:132)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:276)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:103)
	at org.apache.jasper.runtime.PageContextImpl.doForward(PageContextImpl.java:712)
	at org.apache.jasper.runtime.PageContextImpl.forward(PageContextImpl.java:683)
	at i2p.bote.jsp.index_jsp._jspService(index_jsp.java:55)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)
	at i2p.bote.web.CharsetFilter.doFilter(CharsetFilter.java:38)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1484)
	at net.i2p.servlet.filters.XSSFilter.doFilter(XSSFilter.java:28)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:255)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:370)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
	at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
	at java.lang.Thread.run(Thread.java:701)

Migrated from https://trac.i2p2.de/ticket/1644
```json
{
    "status": "closed", 
    "changetime": "2015-11-01T21:19:12", 
    "description": "I2P version: 0.9.21-0\nJava version: Sun Microsystems Inc. 1.6.0_36 (OpenJDK Runtime Environment 1.6.0_36-b36)\nWrapper version: 3.5.15\nServer version: 8.1.17.v20150415\nServlet version: Jasper JSP 2.1 Engine\nPlatform: Linux i386 3.2.0-23-generic\nProcessor: Athlon 5350 APU (athlon64)\nJbigi: Locally optimized native BigInteger library loaded from file\nEncoding: UTF-8\nCharset: UTF-8\n\nError is\nHTTP ERROR 500\n\nProblem accessing /i2pbote/index.jsp. Reason:\n\n    Server Error\n\nCaused by:\n\njava.lang.IllegalArgumentException: Illegal pattern character 'X'\n\tat java.text.SimpleDateFormat.compile(SimpleDateFormat.java:769)\n\tat java.text.SimpleDateFormat.applyPattern(SimpleDateFormat.java:2023)\n\tat org.apache.taglibs.standard.tag.common.fmt.FormatDateSupport.doEndTag(FormatDateSupport.java:164)\n\tat org.apache.jsp.tag.web.printDate_tag._jspx_meth_fmt_005fformatDate_005f0(printDate_tag.java:254)\n\tat org.apache.jsp.tag.web.printDate_tag._jspx_meth_c_005fif_005f0(printDate_tag.java:170)\n\tat org.apache.jsp.tag.web.printDate_tag.doTag(printDate_tag.java:130)\n\tat i2p.bote.jsp.folder_jsp._jspx_meth_ib_005fprintDate_005f0(folder_jsp.java:2826)\n\tat i2p.bote.jsp.folder_jsp._jspx_meth_c_005fforEach_005f0(folder_jsp.java:1941)\n\tat i2p.bote.jsp.folder_jsp._jspx_meth_ib_005frequirePassword_005f0(folder_jsp.java:346)\n\tat i2p.bote.jsp.folder_jsp._jspService(folder_jsp.java:132)\n\tat org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:848)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)\n\tat org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:276)\n\tat org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:103)\n\tat org.apache.jasper.runtime.PageContextImpl.doForward(PageContextImpl.java:712)\n\tat org.apache.jasper.runtime.PageContextImpl.forward(PageContextImpl.java:683)\n\tat i2p.bote.jsp.index_jsp._jspService(index_jsp.java:55)\n\tat org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:848)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)\n\tat i2p.bote.web.CharsetFilter.doFilter(CharsetFilter.java:38)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1484)\n\tat net.i2p.servlet.filters.XSSFilter.doFilter(XSSFilter.java:28)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:255)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:370)\n\tat org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)\n\tat org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)\n\tat org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)\n\tat org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)\n\tat org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)\n\tat org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)\n\tat org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)\n\tat org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)\n\tat java.lang.Thread.run(Thread.java:701)\n", 
    "reporter": "whonose2", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1446412752855385", 
    "component": "apps/plugins", 
    "summary": "New version of Bote (2015_08_21)  throws an error (Java 1.6)", 
    "priority": "minor", 
    "keywords": "I2P-Bote", 
    "version": "0.9.21", 
    "parents": "", 
    "time": "2015-08-21T13:39:37", 
    "milestone": "0.9.23", 
    "owner": "str4d", 
    "type": "defect"
}
```

**Comment by str4d** [comment]
Trac update at 20150822T02:56:53: 
* zzz commented:

> Same issue as str4d previously fixed in net.i2p.router.news.RFC3339Date ?


* zzz changed component from "unspecified" to "apps/plugins"
* zzz changed owner from "" to "str4d"
* zzz changed status from "new" to "assigned"

**Comment by str4d** [comment]
Trac update at 20150822T03:20:02: str4d commented:

> I knew I'd seen it somewhere before. Thanks for the reminder :) Technically though you had already fixed it by including Java 7 checks; I just extended them to Android (its SDF doesn't support XXX at any API).
> 
> The cause of the error is the `X` at the end of the pattern string in `src/main/webapp/WEB-INF/tags/printDate.tag` used to fill the HTML5 `<time>` element's `datetime` attribute. Surrounding it with a Java 7 check (and providing an alternative for pre-Java 7) should fix the error.

**Comment by str4d** [comment]
Trac update at 20150919T12:37:05: str4d changed keywords from "Bote" to "I2P-Bote"

**Comment by str4d** [comment]
Trac update at 20151101T21:19:12: 
* str4d commented:

> Fixed in `0d6505d2805dff9774d816a899b6463074057c15`.


* str4d changed milestone from "undecided" to "0.9.23"
* str4d changed resolution from "" to "fixed"
* str4d changed status from "assigned" to "closed"
